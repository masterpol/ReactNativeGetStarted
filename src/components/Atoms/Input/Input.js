import React from 'react'
import styled from '@emotion/native'
import PropTypes from 'prop-types'

const Element = styled.TextInput`
  border-radius: 10px;
  border: 1px solid ${({ theme }) => theme.colors.primary};
  padding: 5px 20px;
`

const Input = ({ onChangeText, value, placeholder, ...rest }) => (
  <Element
    onChangeText={onChangeText}
    value={value}
    placeholder={placeholder}
    {...rest}
  />
)

Input.propTypes = {
  onChangeText: PropTypes.func.isRequired,
  placeholder: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  value: PropTypes.oneOfType([PropTypes.string, PropTypes.number]).isRequired,
}

export default Input
