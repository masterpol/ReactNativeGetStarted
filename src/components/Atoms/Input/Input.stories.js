import React, { useState } from 'react'
import { storiesOf } from '@storybook/react-native'
import Theme from '_themes'
import styled from '@emotion/native'
import Input from './Input'

const Container = styled.View`
  padding: 0 20px;
`

const InputHandler = () => {
  const [value, setValue] = useState('')

  return (
    <Input
      onChangeText={d => setValue(d)}
      value={value}
      placeholder="General Input"
    />
  )
}

storiesOf('Input', module).add('general', () => (
  <Theme>
    <Container>
      <InputHandler />
    </Container>
  </Theme>
))
