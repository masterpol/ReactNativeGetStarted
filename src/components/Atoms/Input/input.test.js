import React from 'react'
import Input from './Input'
import Theme from '_themes'
import { fireEvent } from '@testing-library/react-native'

describe('Input', () => {
  let props

  beforeEach(() => {
    props = {
      onChangeText: jest.fn(),
      value: '',
      placeholder: 'placeholder',
      testID: 'input-test',
    }
  })

  describe('structure', () => {
    it('should match the snapshot for Input', () => {
      const wrapper = render(
        <Theme>
          <Input {...props} />
        </Theme>,
      )
      expect(wrapper).toMatchSnapshot()
    })
  })

  describe('actions', () => {
    it('should call onChange action', () => {
      const { getByTestId } = render(
        <Theme>
          <Input {...props} />
        </Theme>,
      )

      const text = 'new value'

      fireEvent.changeText(getByTestId('input-test'), text)

      expect(props.onChangeText).toBeCalledWith(text)
    })
  })
})
