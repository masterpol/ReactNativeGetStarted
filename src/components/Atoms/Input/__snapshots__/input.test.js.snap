// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input structure should match the snapshot for Input 1`] = `
<TextInput
  onChangeText={[MockFunction]}
  placeholder="placeholder"
  style={
    Array [
      Object {
        "borderBottomLeftRadius": 10,
        "borderBottomRightRadius": 10,
        "borderColor": "#2196f3",
        "borderStyle": "solid",
        "borderTopLeftRadius": 10,
        "borderTopRightRadius": 10,
        "borderWidth": 1,
        "paddingBottom": 5,
        "paddingLeft": 20,
        "paddingRight": 20,
        "paddingTop": 5,
      },
      undefined,
    ]
  }
  testID="input-test"
  value=""
/>
`;
